pkg/apis/numaflow/v1alpha1/isbsvc_types.go

```diff
@@ -160,3 +147,13 @@ func (isbsvc *InterStepBufferServiceStatus) IsHealthy() bool {
 	}
 	return isbsvc.IsReady()
 }
+
+// MarkChildrenResourceNotHealthy marks the children resources as not healthy
+func (isbsvc *InterStepBufferServiceStatus) MarkChildrenResourceNotHealthy(reason, message string) {
+	isbsvc.MarkFalse(ISBSvcConditionChildrenResourcesHealthy, reason, message)
+}
+
+// MarkChildrenResourceHealthy marks the children resources as healthy
+func (isbsvc *InterStepBufferServiceStatus) MarkChildrenResourceHealthy(reason, message string) {
```

Comment:
when marking healthy, do we need to provide the reason and message? I don't see them in markDeployed and markConfigured.

Reply:
For `kind: Deployment` in k8s is providing `reason` and `message` in healthy case as well as example below and here it's checking the Deployment, Statefulset or Pod status only, So have followed the same.

```
  conditions:
  - lastTransitionTime: "2024-07-25T15:39:30Z"
    lastUpdateTime: "2024-07-25T15:39:30Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
```